### PR TITLE
Overhaul download_files and support unzipping

### PR DESCRIPTION
**Pull request by Yun-Wu**
#96 

1. Add unittest for downloading data from Github/Google Drive/Dropbox
2. Support unzipping 
3. Fix unittest discovery failure
4. Update environment.yml to include new dependencies

### CODE REVIEW

File: tests/test_io_utils.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        output_dir,
+        overwrite_existing=True,
+    )
+
+    data = pd.read_csv(os.path.join(output_dir, "movbank_data.csv"))
+    assert len(data) > 0
+
+
+def test_download_file_dropbox_json():
+    URL = "https://www.dropbox.com/scl/fi/qaw3krcsnot69x94mdfxy/config.json?rlkey=zdmipl2la7rplgl218vc13end&dl=1"
```
Comment (walljcg) [inline tests/test_io_utils.py:48]: Think we should include code to automatically change dl=0 to dl=1 to help out users from having to modify a dropbox url themselves.

File: tests/test_io_utils.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    )
+
+    data = pd.read_csv(os.path.join(output_dir, "download_data.csv"))
+    assert len(data) > 0
+
+
+def test_download_file_gdrive():
+    output_dir = "tests/test_output"
+    ecoscope.io.download_file(
+        "https://drive.google.com/uc?export=download&id=1-AQ9_oacUCcAaiZ6SWU77hZWp1oArQw6",
```
Comment (walljcg) [inline tests/test_io_utils.py:26]: This url string differs from what a user gets when they click on a GDrive file and select 'Copy Link'. Can we transform from that into what's needed here for downloading? E.g., https://docs.google.com/document/d/1EeBSLv7LOr6maTTVY8T08oD0zbsMb9iUyrAjetIWYUo/edit?usp=sharing

File: tests/test_io_utils.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        os.path.join(output_dir, "download_data.csv"),
+        overwrite_existing=True,
+    )
+
+    data = pd.read_csv(os.path.join(output_dir, "download_data.csv"))
+    assert len(data) > 0
+
+
+def test_download_file_gdrive_zip():
```
Comment (walljcg) [inline tests/test_io_utils.py:35]: Is it possible to combine with the above function so that it works for both non-zip and zip files?

**Review by walljcg — CHANGES_REQUESTED** [review]

File: tests/test_io_utils.py (hunk 3 above)
Comment (Yun-Wu) [reply tests/test_io_utils.py:47]: It works for both non-zip and zip files. Here are just two different unittests

File: tests/test_io_utils.py (hunk 2 above)
Comment (Yun-Wu) [reply tests/test_io_utils.py:38]: done

**Review by ericgitonga — COMMENTED** [review]
All tests pass fine.

**Pull request merged**